text/0086-sentry-bundler-plugins-api.md

```diff
@@ -0,0 +1,103 @@
+  headers?: Record<string, string>; // Headers added to every outgoing network request.
+  debug?: boolean; // default: false - Whether to print debug information.
+  silent?: boolean; // default: false - Whether to print anything at all.
+  errorHandler?: (err: Error) => void; // when provided will swallow errors unless the provided function throws itself - By default any error will stop compilation to abort
+  telemetry?: boolean; // default: true - whether to send telemetry to Sentry
+
+  // General configuration for source maps uploading - if not provided, source map uploading will be disabled
+  sourcemaps?: {
+    jsAssets: string | string[]; // globs pointing to the javascript files that should be uploaded to Sentry with their source maps - these are the built assets and not the source files
+    ignore?: string | string[]; // globs to exclude javascript files from being uploaded (will also not upload their source maps)
+    sourcesRoot?: string | null; // default: process.cwd() - We rewrite the `sources` field in source maps to point to sources relative to this value. Set to `null` to disable this rewriting.
+  };
+
+  // General configuration for release creation and release injection - we decouple this from the debug ID source maps upload because they're technically not related anymore.
+  // If not provided, this option will default to its default values.
+  release?: {
+    name?: string; // if not provided - Sentry CLI will try to guess a name (ie. git sha, cloud provider env vars, etc.)
+    inject?: boolean; // default: true - whether to inject the release into bundles
+    create?: boolean; // default: true - if set to `false` will not create a release in Sentry, no matter what other options are set - the release value will still be injected though.
+    finalize?: boolean; // default: true
+    dist?: string; // Can be used to segment releases further
+    vcsRemote?: string; // default: 'origin' - Version control system remote name
+    setCommits?: SetCommitsOptions;
+    deploy?: DeployOptions;
+    legacySourcemaps?: {
+      // Pre-debugid way of uploading source maps - for users that for whatever reason cannot inject code into their bundles
+      include: string | IncludeEntry | Array<string | IncludeEntry>;
+      cleanArtifacts?: boolean; // default: false - Wether to delete artifacts previously uploaded to the release
+    };
+  };
+
+  _experiments: {}; // whatever
+}
+
+// Identical to status-quo. See https://github.com/getsentry/sentry-webpack-plugin/blob/2b7d274a7355f0d27a431b2c20c37c9786bbe4cb/README.md for more information.
+interface IncludeEntry {
+  // ...
+}
+
+// Identical to status-quo. See https://github.com/getsentry/sentry-webpack-plugin/blob/2b7d274a7355f0d27a431b2c20c37c9786bbe4cb/README.md for more information.
+interface SetCommitsOptions {
+  // ...
+}
+```
+
+Some removed options:
+
+- `dryRun` - Instead, just not use the plugin.
```

Comment:
Hm, maybe we should keep it. I can see how it is useful. I just want it to be clean - I guess by making every side-effect a no-op and emitting a message instead.

The thing is that some chains of execution don't make sense in dry run mode anymore: e.g. If we don't inject debug IDs into a bundle due to dry run, the debug ID source map upload step will emit a warning because a file doesn't have a debug id injected.

Reply:
It might make sense to introduce just a top level flag to the plugin to enable/disable it, so that users don't have to conditionally add the plugin, DX feels a little better with that.

Reply:
> IIRC the dry run would for example print the chosen release, right?

Yes, but for printing debug information we already have the `debug` option.

> Maybe as a user, I want to know what release value the plugin/cli chose w/o uploading anything yet?

My view on this is that uploading source maps and bundling your project is **very** low stakes. You can delete releases and source maps with ease.

I see a use for `dryRun` options in tooling where you have a chance of majorly messing something up (e.g. deployment CLIs like terraform). Here it feels overkill.

Right now, I am still leaning towards deleting it for now, since it comes with maintenance effort that scales linearly with new features. I agree with the `disabled` flag though.